Userland/Libraries/LibGfx/ImageFormats/BMPLoader.cpp

```diff
@@ -1372,7 +1373,14 @@ static ErrorOr<void> decode_bmp_pixel_data(BMPLoadingContext& context)
             case 24: {
                 if (!streamer.has_u24())
                     return Error::from_string_literal("Cannot read 24 bits");
-                context.bitmap->scanline(row)[column++] = streamer.read_u24();
+
+                u32 pixel = streamer.read_u24();
+                u8 b = (pixel & 0xFF0000) >> 16;
+                u8 g = (pixel & 0x00FF00) >> 8;
+                u8 r = (pixel & 0x0000FF);
+
+                u32 rgbx_pixel = (r << 16) | (g << 8) | b | 0xFF000000;
+                context.bitmap->scanline(row)[column++] = rgbx_pixel;
```

Comment:
Not sure why this code is more complicated than what we already have in #1167 , but we'd now also have to rehash the conversation we had there: why would you set the alpha channel for a bitmap format that has no alpha channel?

Reply:
Check your words there
> I agree with you that using RGB(A) as an output type makes more sense for Ladybird since Skia (and possibly other rendering libraries as well) natively supports both the version with and without the alpha channel.

This Pull Request reorders bytes to pass RGBx as output type. Without alpha channel. Last byte is a spacer, it does nothing important and is not rendered as Alpha channel. This is why this code is more complicated. It is written according to the discussion you had at https://github.com/LadybirdBrowser/ladybird/pull/1167 and on @ LucasChollet and your point that RGB(A) should be passed to Skia.

Reply:
It's not possible to pass the image as is to Skia. Skia does not support BGRx8888. So we either add alpha channel as in https://github.com/LadybirdBrowser/ladybird/pull/1167 or we reorder bytes as in this PR.

Reply:
I deleted the 0xFF000000 part to make code less confusing. Now, the spacer is undefined (perhaps some default value or 0)

Reply:
I also want to mention that there's no other image format that we do not append alpha channel to. With this PR, BMP becomes the first one. So your question "why would you set the alpha channel for a bitmap format that has no alpha channel?" applied to any absolutely any image we render, be it PNG, JPEG, TIFF etc

Reply:
I'm only talking about the referenced code, not the other changes in this PR which seem good! If we use RGBx, we should not care about the alpha channel.

Reply:
It seems that my changes to remove the extra to_skia_color_type broke the CI. Do you have any ideas why? It builds fine on my Mac with AppKit Chrome. I could revert the change and keep to_skia_color_type as it was originally, but I’d prefer not to have duplicated code. Duplicated code can be confusing, especially since the functions with the same name are used in both the cacher and renderer. This can lead to unexpected behavior if they differ.

Reply:
hm, no idea why the CI fails with ``LibGfx/PainterSkia.h:12:10: error: 'core/SkImageInfo.h' file not found``.

Reply:
The point is, Skia _should_ be available at compile time and that issue is not reproducible on my local machine..